### DISCUSSION TITLE
Adding data-pagefind-sort removed pages.

### DISCUSSION THREAD

**Discussion by huphtur** [Pagefind/pagefind#255] [Q&A]
Have 2 types of pages on my 11ty site: collection (blog) pages, and one-off pages (eg an about page).

Added `data-pagefind-body` to `<main>` and all the pages get indexed, which is great.

I wanted to sort the blog posts by date, so added `data-pagefind-sort="date"` and `sort: { date: "asc" }`. My blog posts are indeed sorted, but now my one-off pages are not shown in the search result anymore.

What am I doing wrong here?

**Comment by bglw** [comment]
Currently what you're seeing is intentional, where using a sort acts as an impromptu filter as well.

What would be your desired behavior here? If you search for a term and sort the results by `date`, the one-off pages would have to all either appear before or after the blog posts that do have a date, in some undetermined order.

**Reply by huphtur** [reply]
Ah, that makes sense. If I wanted to show then before the blog post what would be the best action to handle that? I could add today's date to the one-off pages, but assume that's a bit wonky.

**Reply by bglw** [reply]
Hmm, if you're using the default UI I can't think of a great way to achieve that right now. If you're using the JS search API, I would recommend just running two searches: one with the sort and one with a filter to pages.

I think ideally this is a feature I add to the sorting behavior that can be configured in some way.

**Reply by bglw** [reply]
What would you expect to see when changing the date sort between `asc` and `desc`? Would untagged results stay pinned to the top, or would they move to be at the end of the search results? (This weird interaction is the reason I currently omit them)
